dcos/prometheus-dcos.yml

```diff
@@ -23,21 +23,9 @@ scrape_configs:
     replacement: $1
 
   metric_relabel_configs:
-  # remove explicit linkerd proxy ports
-  - source_labels: [__name__]
-    action:        replace
-    target_label:  __name__
-    regex:         rt:([^:]+):srv:0_0_0_0:[^:]+:(.*)
-    replacement:   rt:$1:srv:0_0_0_0:port:$2
-  # write service names into label {service:"app1"}
-  - source_labels: [__name__]
+  # write service names into service label {service:"app1"}
+  - source_labels: [dst_id]
     action:        replace
     target_label:  service
-    regex:         rt:[^:]+:dst:id.*:_:io_l5d_marathon:([^:]+):.*
+    regex:         .*/([^/]+)
```

Comment:
A few days ago in slack somebody mentioned that linkerd-viz wasn't properly handling metrics of the format:

```
rt:incoming:dst:id:_:io_l5d_localhost:_:io_l5d_marathon:system:cadvisor:dispatcher:serial:queue_size 0.0
```

Specifically, we're setting service_name to `cadvisor`, and dropping `dispatcher:serial`. I think we can actually fix that bug as part of this change, by tweaking this regex slightly.